src/alchemlyb/tests/parsing/test_namd.py

```diff
@@ -36,13 +48,183 @@ def test_u_nk_idws():
+    using caller-supplied functions.
+
+    Parameters
+    ----------
+    fepout_in: str
+        Path to fepout file to be modified. This file will not be overwritten.
+    
+    params: list of tuples
+        For each tuple, the first element must be a str that will be passed to
+        startswith() to identify the line(s) to modify (e.g. "#NEW"). The
+        second element must be a function that accepts a list of strs which is
+        the output of running split() on the identified line and returns
+        a modified list of tokens that will be reassembled into the line to be
+        output.
+
+    Returns
+    -------
+    The name of a temporary file which the caller must unlink.
+    """
+
+    fh, fepout_out = mkstemp(suffix='.fepout.bz2', prefix='restarted_')
+    close(fh)
+    with bz2.open(fepout_out, 'wt') as f_out:
+        with bz2.open(fepout_in, 'rt') as f_in:
+            for line in f_in:
+                for prefix, func in params:
+                    if line.startswith(prefix):
+                        line = ' '.join(func(line.split())) + '\n'
+                f_out.write(line)
+    return fepout_out
+
+
+@pytest.fixture
+def restarted_dataset_inconsistent():
+    """Returns intentionally messed up dataset."""
+    # We explicitly call load_restarted() again because we are modifying the
+    # singleton dataset object and don't want to break other tests
+    dataset = load_restarted()
+    filenames = dataset['data']['both']
+
+    changed = False
+    def func_free_line(l):
+        if float(l[7]) >= 0.7 and float(l[7]) < 0.9:
+            l[7] = str(float(l[7]) + 0.0001)
+            changed = True
+        return l
+
+    temp_fnames = []
+    for i in range(len(filenames)):
+        fname = _corrupt_fepout(filenames[i], [('#Free', func_free_line)])
+        dataset['data']['both'][i] = fname
+        temp_fnames.append(fname)
+        # Only actually modify one window so we don't trigger the wrong exception
+        if changed is True:
+            break
+
+    yield dataset
+    for fname in temp_fnames:
+        unlink(fname)
+
+
+@pytest.fixture
+def restarted_dataset_toomany_lambda2():
+    """Returns intentionally messed up dataset, where there are too many lambda2 values for a
+    given lambda1."""
+    dataset = load_restarted()
+    filenames = dataset['data']['both']
+
+    # For the same l1 and lidws we retain old lambda2 values thus ensuring a collision
+    def func_new_line(l):
+        l[LAMBDA1_IDX_NEW] = '0.2'
+        if len(l) > 9 and l[9] == 'LAMBDA_IDWS':
+            l[LAMBDA_IDWS_IDX_NEW] = '0.1'
+        return l
+
+    def func_free_line(l):
+        l[LAMBDA1_IDX_FREE] = '0.2'
+        return l
+
+    temp_fnames = []
+    for i in range(len(filenames)):
+        fname = _corrupt_fepout(filenames[i], [('#NEW', func_new_line), ('#Free', func_free_line)])
+        dataset['data']['both'][i] = fname
+        temp_fnames.append(fname)
+
+    yield dataset
```

Comment:
see above, just return, and don't clean up (assuming everything uses `tmpdir`/`tmp_path`)